nameko/events.py

```diff
@@ -101,64 +101,77 @@ def dispatch(event_type, event_data):
-
-        If `reliable_delivery` is true, events will be held in the queue
-        until there is a handler to consume them. Defaults to true.
-
-        Raises an ``EventHandlerConfigurationError`` if the ``handler_type``
-        is set to ``BROADCAST`` and ``reliable_delivery`` is set to ``True``.
+        called ``source_service``.
+
+        :Parameters:
+            source_service : str
+                Name of the service that dispatches the event
+            event_type : str
+                Type of the event to handle
+            handler_type : str
+                Determines the behaviour of the handler in a cluster:
+
+                - ``events.SERVICE_POOL``:
+
+                    Event handlers are pooled by service type and method,
+                    and one service instance from each pool receives the
+                    event. ::
+
+                                   .-[queue]- (service X handler-meth-1)
+                                  /
+                        exchange o --[queue]- (service X handler-meth-2)
+                                  \
+                                   \          (service Y(inst. 1) handler-meth)
+                                    \       /
+                                     [queue]
+                                            \
+                                              (service Y(inst. 2) handler-meth)
+
+                - ``events.SINGLETON``:
+
+                    Events are received by only one registered handler,
+                    regardless of service type. If requeued on error, they may
+                    be handled by a different service instance. ::
+
+                                               (service X handler-meth)
+                                             /
+                        exchange o -- [queue]
+                                             \
+                                               (service Y handler-meth)
+
+                - ``events.BROADCAST``:
+
+                    Events will be received by every handler. Events are
+                    broadcast to every service instance, not just every service
+                    type - use wisely! ::
+
+                                    [queue]- (service X(inst. 1) handler-meth)
+                                  /
+                        exchange o - [queue]- (service X(inst. 2) handler-meth)
+                                  \
+                                    [queue]- (service Y handler-meth)
+
+            requeue_on_error : bool  # TODO: defined by Consumer actually..
+                If true, handlers will return the event to the queue if an
+                error occurs while handling it. Defaults to False.
+            reliable_delivery : bool
+                Is true, events will be held in the queue until there is a
+                handler to consume them. Defaults to True.
+            sensitive_variables : string or tuple of strings
```

Comment:
Perhaps not. I was matching the interface of ``expected_exceptions``.  A common usecase would be something like:

``` python
class Service(object):
    @rpc(sensitive_variables=("foo",), expected_exceptions=FooError)
    def method(self, foo, bar):
        pass
```

The parenthesis seem slightly superfluous given we don't require them for ``expected_exceptions``.